### ISSUE TITLE
Add site.Store

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#13021]
Note that `site.Store != site.Home.Store`.

**Comment by dvdksn** [comment]
Should there be a `SHORTCODE.Store` too? That's still using `Scratch`.

**Comment by bep** [comment]
>Should there be a SHORTCODE.Store too? That's still using Scratch.

That's true. For consistency I think we should do a "soft rename". For context, the main reason I'm proposing this now is that I'm about to finish the `js.Batch` template func, and the signature of that one looks like this:

```go
func (j *JSFuncs) Batch(id string, store *maps.Scratch) (esbuild.Batcher, error) {
}
```

Meaning you need to provide a "storage scope", e.g.

```
{{ with js.Batch "adminscripts" site.Store }}
...
{{ end }}
```

And while it's true that we never "clear a store", the store of a given `Page` will get recreated if the page itself is recreated (e.g. by editing the content while the server is running). `hugo.Store` and `site.Store` will only be cleared on full rebuilds (which we currently do when you edit a config file).

/cc @jmooring
